src/openrct2-ui/windows/RideConstruction.cpp

```diff
@@ -4845,6 +4845,8 @@ namespace OpenRCT2::Ui::Windows
         uint16_t x, y, properties;
         SelectedLiftAndInverted liftHillAndInvertedState{};
 
+        auto intent = Intent(INTENT_ACTION_RIDE_CONSTRUCTION_UPDATE_PIECES);
+        ContextBroadcastIntent(&intent);
```

Comment:
The function can return several times before reaching this now, is that intended?

Maybe we want to use RAII to make sure to broadcast this before the function closes (?)

Reply:
> The function can return several times before reaching this now, is that intended?

It is correct, but it previously did exactly the same. `WindowRideConstructionUpdateStateGetTrackElement()`, which is called on the next line, previously broadcast it as the first thing it did. I just moved it here. I did that because I needed the output from `WindowRideConstructionUpdateStateGetTrackElement()` somewhere else too, but without the broadcast.

Put simply, I changed():
```C++
spam();
eggs();
auto updated_element = WindowRideConstructionUpdateStateGetTrackElement();

...

std::pair<bool, TrackElemType> WindowRideConstructionUpdateStateGetTrackElement()
{
    auto intent = Intent(INTENT_ACTION_RIDE_CONSTRUCTION_UPDATE_PIECES);
    ContextBroadcastIntent(&intent);

    ...
}
```

To this:

```C++
spam();
eggs();
auto intent = Intent(INTENT_ACTION_RIDE_CONSTRUCTION_UPDATE_PIECES);
ContextBroadcastIntent(&intent);
auto updated_element = WindowRideConstructionUpdateStateGetTrackElement();

...

std::pair<bool, TrackElemType> WindowRideConstructionUpdateStateGetTrackElement()
{
    ...
}
```

Reply:
Thanks for both explanations!